### PR TITLE
Use stringbuilder in ecma_raise_standard_error_with_format

### PR DESCRIPTION
**Pull request by szilagyiadam**
JerryScript-DCO-1.0-Signed-off-by: Adam Szilagyi aszilagy@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 1]
```diff
@@ -257,7 +257,11 @@ ecma_raise_standard_error_with_format (ecma_standard_error_t error_type, /**< er
 {
   JERRY_ASSERT (format != NULL);
 
-  ecma_string_t *error_msg_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+  ecma_string_t *empty_str_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+
+  ecma_stringbuilder_t builder = ecma_stringbuilder_create_from (empty_str_p);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-exceptions.c]: You can use `ecma_stringbuilder_create` .

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 2]
```diff
@@ -257,7 +257,11 @@ ecma_raise_standard_error_with_format (ecma_standard_error_t error_type, /**< er
...
+
+  ecma_deref_ecma_string (empty_str_p);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-exceptions.c]: This deref is not needed at all.

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 3]
```diff
@@ -320,17 +319,11 @@ ecma_raise_standard_error_with_format (ecma_standard_error_t error_type, /**< er
...
-    error_msg_p = ecma_append_chars_to_string (error_msg_p,
-                                               chars_p,
-                                               chars_size,
-                                               lit_utf8_string_length (chars_p, chars_size));
+    ecma_stringbuilder_append_raw (&builder, (lit_utf8_byte_t *) start_p, (lit_utf8_size_t) (end_p - start_p));
   }
 
-  ecma_object_t *error_obj_p = ecma_new_standard_error_with_message (error_type, error_msg_p);
-  ecma_deref_ecma_string (error_msg_p);
+  ecma_object_t *error_obj_p = ecma_new_standard_error_with_message (error_type,
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-exceptions.c]: The reference count of the finalized string should be decreased.

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 4]
```diff
@@ -257,7 +257,9 @@ ecma_raise_standard_error_with_format (ecma_standard_error_t error_type, /**< er
 {
   JERRY_ASSERT (format != NULL);
 
-  ecma_string_t *error_msg_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+  ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+
+  ecma_stringbuilder_append_magic (&builder, LIT_MAGIC_STRING__EMPTY);
```
Comment (dbatyai) [inline jerry-core/ecma/operations/ecma-exceptions.c]: There's no reason to append the empty string, the result will remain the same.

**Pull request merged**